### ISSUE TITLE
How do we feel about dependencies?

### ISSUE THREAD

**Issue by Zulko** [Global-Biofoundries-Alliance/SynBioPython#7]
Few dependencies is a nice trait in a library. Biopython's only dependency seems to be Numpy.

I am asking because Plateo depends on a few heavy libraries (the core can be standalone, but dealing with spreadsheets is easier with Pandas/xlrd), and I expect other modules may depend on at least Biopython and a few others.

**Comment by neilswainston** [comment]
I’d also suggest keeping dependencies to a minimum, but am fairly relaxed about dependencies on ubiquitous libraries such as numpy, scipy, Pandas, etc.

**Comment by zoltuz** [comment]
I agree, being mindful about the dependencies will pay dividends in the long run for sure. But as long as they are pip installable it should be fine. 

Alternatively, we can put a script together which creates/updates a (docker) container that encapsulates all (not just python) dependences. 
https://docs.docker.com/compose/gettingstarted/

Recently, Netflix released its in-house software called Metaflow. One of the features of Metaflow is the capability to run/test out the python code on different external library versions to weed out dependency related issue (in their case, mostly in the result of the computation, e.g. output of a data science model).
https://docs.metaflow.org/metaflow/dependencies
